python/ray/dag/compiled_dag_node.py

```diff
@@ -1887,6 +1887,64 @@ async def execute_async(
         self._execution_index += 1
         return fut
 
+    def visualize(self, filename="compiled_dag", format="png", view=False):
+        """
+        Visualize the compiled DAG using Graphviz.
+
+        Args:
+            filename: The name of the output file (without extension).
+            format: The format of the output file (e.g., 'png', 'pdf').
+            view: Whether to open the file with the default viewer.
+        """
+        import graphviz
+        from ray.dag import (
+            InputAttributeNode,
+            InputNode,
+        )
+
+        dot = graphviz.Digraph(name="CompiledDAG", format=format)
+
+        # Add nodes with task information
+        for idx, task in self.idx_to_task.items():
+            dag_node = task.dag_node
+
+            # Initialize the label
+            label = f"Task {idx}\n"
+
+            # Handle different types of dag_node
+            if isinstance(dag_node, InputNode):
+                label += "InputNode"
+            elif isinstance(dag_node, InputAttributeNode):
+                label += f"InputAttributeNode[{dag_node.key}]"
+            elif hasattr(dag_node, "get_method_name"):
+                method_name_attr = getattr(dag_node, "get_method_name")
+                if callable(method_name_attr):
+                    method_name = method_name_attr()
+                else:
+                    method_name = method_name_attr
+
+                # Get actor ID if applicable
+                actor_handle = dag_node._get_actor_handle()
+                if actor_handle:
```

Comment:
I see some non-driver nodes don't have actors, why is that?

Reply:
This is due to multi-returns. I think we should make the visualized nodes more understandable by end users.

Reply:
One idea is to use rectangles instead of ovals for these special nodes.